__TESTS__/unit/actions/Overlay.test.ts

```diff
@@ -0,0 +1,30 @@
+import {fill} from "../../../src/actions/resize/Resize";
+import {auto, best} from "../../../src/actions/quality/Quality";
+import {imageLayer} from "../../../src/actions/layers/overlay";
+import {imageSource} from "../../../src/actions/layers/Sources";
+import TransformableImage from "../../../src/transformation/TransformableImage";
+
+describe('Tests for overlay actions', () => {
+  it('Parses an overlay with an imageSource', () => {
+    const tImage = new TransformableImage('sample');
+    tImage
+      .setConfig({
+        cloud: {
+          cloudName:'demo'
+        }
+      })
+      .resize(fill(200, 200))
+      .quality(best())
+      .overlay(
+        // imageLayerAction is what registers on the transformation(which has toString() method)
+        // imageSource is the parameter for imageLayerAction
+        imageLayer( imageSource('sample')
```

Comment:
I'm not sure how you add the rest of the layer-action components here (mainly position and blendmode). I'm not saying the PoC needs the full implementation, but we do need to verify it's doable and ends up with a decent API once those are added.